docs: document replica scheduling constraints (#21090)
MaterializeInc/materialize: doc/user/content/sql/create-cluster-replica.md

{{% replica-options %}}

{{< note >}}

> If you do not specify an availability zone, Materialize will schedule
> the replica in an arbitrary availability zone. Guarantees on how replicas

Reviewer: ```suggestion
If you do not specify an availability zone, Materialize will assign the replica
to an arbitrary availability zone. Guarantees on how replicas
```
